src/Provider/AbstractProvider.php

```diff
@@ -686,18 +702,26 @@ protected function parseResponse(ResponseInterface $response)
             return $this->parseJson($content);
         } catch (UnexpectedValueException $e) {
             if (strpos($type, 'json') !== false) {
-                throw $e;
+                throw $this->mayThrowResponseParsingException
+                    ? new ResponseParsingException($response, $content, $e->getMessage(), $e->getCode())
```

Comment:
Following my other comments about adding `$previous` to the constructor for `ResponseParsingException`, pass `$e` here as the 5th parameter.